celery/platforms.py

```diff
@@ -411,8 +417,8 @@ def after_chdir_do():
         # we need to know that we have access to the logfile.
         logfile and open(logfile, 'a').close()
         # Doesn't actually create the pidfile, but makes sure it's not stale.
-        if pidfile:
-            _create_pidlock(pidfile).release()
+        if pidfile and not _can_create_pidlock(Pidfile(pidfile)):
```

Comment:
You could possibly extract this branch to a new function, something like:

```python
def _ensure_not_stale(pidfile):
    if pidfile:
      ...
```
This way you could even make the comment redundant :smile: !

Reply:
Good point